### DISCUSSION TITLE
I'm considering using vue-query for a Nuxt3 project. I would appreciate answers to some questions and your opinion.

### DISCUSSION THREAD

**Discussion by hirotaka** [nuxt/nuxt#23294] [Questions]
Hi, everyone!

You may organize the application by splitting the components into multiple pieces as the application grows. 
If you want to refer to values obtained from the same API in those multiple components, the issue arises regarding where to do the fetch.
In the React community, [@tanstack/query](https://tanstack.com/query/latest) will help with this issue.

An example using `vue-query`, a port of `@tanstack/query` to vue, would look like this:

https://stackblitz.com/edit/nuxt-starter-adptt4?file=pages%2Fproducts%2F%5Bid%5D.vue

The fetch is called twice when displaying product details, but the `vue-query` would handle these multiple requests.

So here's my question:
1. I thought I could do the same with useFetch in Nuxt3. Based on [my previous question](https://github.com/nuxt/nuxt/discussions/22960#discussion-5584725), `useFetch` alone cannot accomplish this process. Am I correct in this understanding?
2. What is the Nuxt3 way to approach this issue without using `@tanstack/query`?
3. Is using `@tanstack/query` in the Nuxt3 project a good idea?

I'm familiar with React and `@tanstack/query`. However, I have no experience with `vue-query` in my Nuxt3 project. I would love to hear your opinion if anyone has used it in a project.

On the other hand, I'm a Nuxt3 newbie, so if you have any suggestions on how to do something better with Nuxt3, I would appreciate it.

**Comment by Hebilicious** [comment]
@hirotaka If you like tanstack (like me), feel free to use this nuxt module https://github.com/Hebilicious/vue-query-nuxt

**Reply by hirotaka** [reply]
Wow, it's great! I appreciate your work.

**Comment by hirotaka** [comment]
I think the blog post in the comment on this issue(https://github.com/nuxt/nuxt/issues/22688#issuecomment-1722944613) may be helpful.
